devops/azure-pipelines/maven-build-test-stage-template.yml

```diff
@@ -0,0 +1,33 @@
+stages:
+- stage: Build_and_Test
+  displayName: ${{ parameters.displayName }}
+  jobs:
+  - job: Build_and_Test
+    steps:
+    - task: SonarCloudPrepare@1
+      inputs:
+        SonarCloud: 'Forge-SonarCloud'
+        organization: '1058-sophia-gavrila-forge'
+        scannerMode: 'Other'
+        # extraProperties: |
+        #   # sonar.branch.name=main 
+        #   sonar.projectKey=1058-Sophia-Gavrila-Forge_Forge-Backend
+        # don't hardcode the branch name like this - it's already wrong for the auto-on PR builds.
+        # sonar scanner discovers the branch name automatically with its git and github integrations, 
+        # so just let it be. -Nick's advice (will check this once we actually get maven pom file)
```

Comment:
if this implementation isn't ready, then why merge it? move it off to it's own branch